Zend/zend_operators.h

```diff
@@ -133,6 +134,18 @@ static zend_always_inline zend_long zend_dval_to_lval_cap(double d)
 }
 /* }}} */
 
+static zend_always_inline bool is_long_compatible(double d) {
```

Comment:
zend_ prefix?

As written this code is also UB and will be flagged by ubsan. I think you may be better served by making this

```
static zend_always_inline bool zend_is_long_compatible(double dval, zend_long lval) {
    return (double) lval == dval;
}
```
and have the caller pass in the `zend_dval_to_lval` result. Also avoids a duplicate calculation.